src/serial/serial_buffer.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Race Capture Pro Firmware
+ *
+ * Copyright (C) 2015 Autosport Labs
+ *
+ * This file is part of the Race Capture Pro fimrware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mem_mang.h"
+#include "printk.h"
+#include "serial.h"
+#include "serial_buffer.h"
+#include "stdutil.h"
+#include "string.h"
+#include "taskUtil.h"
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+static void ctrl_char_strip(char *data)
+{
+        /* Strip all control characters from our messages */
```

Comment:
This was adapted from what existed previously.  It strips any non-visible characters.  It is designed for trailing control characters since that is how our serial rx methods parse lines (on \n)